workflow/executor/resource.go

```diff
@@ -34,7 +35,11 @@ func (we *WorkflowExecutor) ExecResource(action string, manifestPath string, fla
 	cmd := exec.Command("kubectl", args...)
 	log.Info(strings.Join(cmd.Args, " "))
 
-	out, err := cmd.Output()
+	var out []byte
+	err = retry.OnError(retry.DefaultBackoff, argoerr.IsTransientErr, func() error {
```

Comment:
hmmm... what does this do, run the command X many times? 
@sarabala1979 @whynowy  I did not know about `retry.OnError` - better that argowait I think

Reply:
agree with @alexec .  `argowait` should be used.  you can enhance argowait with `onError()` if you want it.

Reply:
sorry - I actually meant the opposite - we should stop using argowait, and use retry.OnError instead

Reply:
Example:
```golang
	err := waitutil.Backoff(defaultRetry,
		func() (bool, error) {
			log.Infof("GCS List bucekt: %s, key: %s", artifact.GCS.Bucket, artifact.GCS.Key)
			client, err := g.newGCSClient()
			if err != nil {
				log.Warnf("Failed to create new GCS client: %v", err)
				return isTransientGCSErr(err), err
			}
			defer client.Close()
			files, err = listByPrefix(client, artifact.GCS.Bucket, artifact.GCS.Key, "")
			if err != nil {
				return isTransientGCSErr(err), err
			}
			return true, nil
		})
```
Becomes
```
	err := retry.OnError(defaultRetry, isTransientGCSErr,
		func() error {
			log.Infof("GCS List bucekt: %s, key: %s", artifact.GCS.Bucket, artifact.GCS.Key)
			client, err := g.newGCSClient()
			if err != nil {
				log.Warnf("Failed to create new GCS client: %v", err)
				return err
			}
			defer client.Close()
			files, err = listByPrefix(client, artifact.GCS.Bucket, artifact.GCS.Key, "")
			if err != nil {
				return err
			}
			return nil
		})
```

Reply:
ok